src/BaseDir.hs

```diff
@@ -0,0 +1,24 @@
+module BaseDir 
+    ( validateDir
+    , baseDirArg
+    )
+    where
+
+import Prologue
+
+import Options.Applicative
+import Path.IO (resolveDir', doesDirExist)
+import System.Exit (die)
+
+-- | Validate that a filepath points to a directory and the directory exists
```

Comment:
I think the type makes it clear, but adding another sentence about how the program exits if the filepath doesn't validate could be helpful